main.cpp

```diff
@@ -113,6 +124,76 @@ Thing 4) chord
     3) be omitted
  */
 
+/*
+Thing 5) Keys
+5 properties:
+    1) is key depressed
+    2) velocity of key pressed (1-127)
+    3) MIDI number of note pressed (1-127)
+    4) Is note legato (note held, while another is played)
+    5) Is keyboard set to monophonic
+3 things it can do:
+    1) play legato
+    2) play chords
+    3) play monophonic
+ */
+
+/*
+Thing 6) Oscillator
+5 properties:
+    1) pitch value of oscillator
+    2) octave setting of oscillator (int 'foot' selection of LFO, 32, 16, 8, 4, 2)
+    3) is oscillator on
+    4) is oscillator controlled by pitch via keys
+    5) waveform of oscillator
+3 things it can do:
+    1) make a sound
+    2) be an LFO
+    3) be a square wave
+ */
+
+/*
+Thing 7) Filter
+5 properties:
+    1) Cutoff frequency
+    2) Resonance
+    3) Attack value
+    4) Decay value
+    5) Release value
+3 things it can do:
+    1) change oscillator cutoff
+    2) change oscillator resonance
+    3) receive host automation
+ */
+
+/*
+Thing 8) Amplitude Envelope
+5 properties:
+    1) Attack value
+    2) Decay value
+    3) Sustain value
+    4) Release value
+    5) Envelope is on
+3 things it can do:
+    1) adjust oscillator attack
+    2) receive host automation
+    3) be bypassed
+ */
+
+/*
+Thing 9) Arpeggiator
+5 properties:
+    1) Octave range value
+    2) Number of note repetitions before moving to next note
+    3) is on
```

Comment:
find something other than `is on`
maybe `sequence number`, which can be represented with an `int`
- [ ] revise